lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.py

```diff
@@ -49,47 +53,158 @@ def createPrimAndAttributes(stage, primPath, mayaReferencePath, mayaNamespace, m
+                    om.MGlobal.displayError(errorMsg)
+                    return Usd.Prim()
+                groupPrimName = Tf.MakeValidIdentifier(checkGroupPrimName)
+
+        # If the group prim was either not provided or empty we use a default name.
+        if not groupPrimName:
+            groupPrimName = getDefaultGroupPrimName(parentPrim, mayaNamespace)
+
+    # When the input is a gateway we cannot have in variant unless group is also given.
+    if isGateway and variantSet and not groupPrimName:
+        errorMsg = getMayaUsdLibString('kErrorCannotAddToProxyShape')
+        om.MGlobal.displayError(errorMsg)
+        return Usd.Prim()
+
+    # Make sure the input Maya Reference prim name doesn't exist already
+    # and validate the input name.
+    # Note: if we are given a group prim to create, then we know that the
+    #       Maya Reference prim name will be unique since it will be the
+    #       only child (of the newly created group prim).
+    checkName = mayaUsd.ufe.uniqueChildName(parentPrim, mayaReferencePrimName) if groupPrim is None else mayaReferencePrimName
     if checkName != mayaReferencePrimName:
-        cmds.error('Maya Reference prim name already exists.')
-        return
+        errorMsgFormat = getMayaUsdLibString('kErrorMayaRefPrimExists')
+        errorMsg = cmds.format(errorMsgFormat, stringArg=(mayaReferencePrimName, ufePathStr))
+        om.MGlobal.displayError(errorMsg)
+        return Usd.Prim()
     validatedPrimName = Tf.MakeValidIdentifier(checkName)
 
     # Extract the USD path segment from the UFE path and append the Maya
     # reference prim to it.
-    ufePath = ufe.PathString.path(ufePathStr)
     parentPath = str(ufePath.segments[1]) if ufePath.nbSegments() > 1 else ''
-    primPath = Sdf.AssetPath(parentPath + '/' + validatedPrimName)
 
     stage = mayaUsd.ufe.getStage(ufePathStr)
 
-    variantSetName = Tf.MakeValidIdentifier(variantSetName)
-    variantName = Tf.MakeValidIdentifier(variantName)
+    # Optionally insert a Group prim as a parent of the Maya reference prim.
+    groupPrim = None
+    if groupPrimName:
+        groupPath = Sdf.AssetPath(parentPath + '/' + groupPrimName)
+        try:
+            groupPrim = stage.DefinePrim(groupPath.path, groupPrimType)
+        except (Tf.ErrorException):
+            groupPrim = Usd.Prim()
+        if not groupPrim.IsValid():
+            errorMsgFormat = getMayaUsdLibString('kErrorCreatingGroupPrim')
+            errorMsg = cmds.format(errorMsgFormat, stringArg=(ufePathStr))
+            om.MGlobal.displayError(errorMsg)
+            return Usd.Prim()
+        if groupPrimKind:
+            model = Usd.ModelAPI(groupPrim)
+            model.SetKind(groupPrimKind)
+
+    if groupPrim:
+        primPath = Sdf.AssetPath(groupPrim.GetPath().pathString + '/' + validatedPrimName)
+    else:
+        primPath = Sdf.AssetPath(parentPath + '/' + validatedPrimName)
```

Comment:
If we inserted a group prim, then adjust the prim path for the Maya Reference prim (to be child of group).